packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormSendEmail.tsx

```diff
@@ -62,48 +107,115 @@ export const WorkflowEditActionFormSendEmail = ({
+    objectNameSingular: 'connectedAccount',
+    filter,
+  });
+
+  let emptyOption: SelectOption<string | null> = { label: 'None', value: null };
+  const connectedAccountOptions: SelectOption<string | null>[] = [];
+
+  accounts.forEach((account) => {
+    const selectOption = {
+      label: account.handle,
+      value: account.id,
+    };
+    if (account.accountOwnerId === currentWorkspaceMember?.id) {
+      connectedAccountOptions.push(selectOption);
+    } else {
+      // This handle the case when the current connected account does not belong to the currentWorkspaceMember
+      // In that case, current connected account email is displayed, but cannot be selected
+      emptyOption = selectOption;
+    }
+  });
+
   return (
-    <WorkflowEditActionFormBase
-      ActionIcon={<IconMail color={theme.color.blue} />}
-      actionTitle="Send Email"
-      actionType="Email"
-    >
-      <StyledTriggerSettings>
-        <Controller
-          name="subject"
-          control={form.control}
-          render={({ field }) => (
-            <TextInput
-              label="Subject"
-              placeholder="Thank you for building such an awesome CRM!"
-              value={field.value}
-              onChange={(email) => {
-                field.onChange(email);
-
-                handleSave();
-              }}
-            />
-          )}
-        />
-
-        <Controller
-          name="body"
-          control={form.control}
-          render={({ field }) => (
-            <TextArea
-              label="Body"
-              placeholder="Thank you so much!"
-              value={field.value}
-              minRows={4}
-              onChange={(email) => {
-                field.onChange(email);
-
-                handleSave();
-              }}
-            />
-          )}
-        />
-      </StyledTriggerSettings>
-    </WorkflowEditActionFormBase>
+    !loading && (
+      <WorkflowEditActionFormBase
+        ActionIcon={<IconMail color={theme.color.blue} />}
+        actionTitle="Send Email"
+        actionType="Email"
+      >
+        <StyledTriggerSettings>
+          <Controller
+            name="connectedAccountId"
+            control={form.control}
+            render={({ field }) => (
+              <Select
+                dropdownId="select-connected-account-id"
+                label="Account"
+                fullWidth
+                emptyOption={emptyOption}
+                value={field.value}
+                options={connectedAccountOptions}
+                callToActionButton={{
+                  onClick: connectAccountFromWorkflow,
+                  Icon: IconPlus,
+                  text: 'Add account',
+                }}
+                onChange={(connectedAccountId) => {
+                  field.onChange(connectedAccountId);
+                  handleSave();
+                  checkConnectedAccountScopes(connectedAccountId);
```

Comment:
What does happen if the user closes the Google tab we opened before completing the process? The account will be wired to the step – `handleSave()` is called before ` checkConnectedAccountScopes()` – but the account wouldn't be set up correctly. Is it possible to recover from this error in the app currently?

Reply:
you are right. In fact I call handleSave() before because we don't have access to the handleSave() method after the google oAuth, so it is not possible to do it. 
In fact it is possible, but it would be a lot of work for something "temporary" as the accounts will have the proper scopes after they have been updated
Also, I think this oAuth triggered at dropdown select is a little bit aggressive, a warning banner would be better in my opinion, will discuss that with @Bonapara